node/bft/storage-service/src/persistent.rs

```diff
@@ -152,66 +179,84 @@ impl<N: Network> StorageService<N> for BFTPersistentStorage<N> {
                 }
                 Ok(None) => {
-                    // Retrieve the missing transmission.
-                    let Some(transmission) = missing_transmissions.remove(&transmission_id) else {
-                        if !aborted_transmission_ids.contains(&transmission_id)
-                            && !self.contains_transmission(transmission_id)
+                    // The transmission is missing from persistent storage.
+                    // Check if it exists in the `missing_transmissions` map provided.
+                    if let Some(transmission) = missing_transmissions.remove(&transmission_id) {
+                        let certificate_ids = indexset! { certificate_id };
+
+                        // Insert into persistent storage.
+                        if let Err(e) =
+                            self.transmissions.insert(transmission_id, (transmission.clone(), certificate_ids.clone()))
                         {
-                            error!("Failed to provide a missing transmission {transmission_id}");
+                            error!("Failed to insert transmission {transmission_id} into storage - {e}");
                         }
-                        continue 'outer;
-                    };
-                    // Prepare the set of certificate IDs.
-                    let certificate_ids = indexset! { certificate_id };
-                    // Insert the transmission and a new set with the certificate ID.
-                    if let Err(e) = self.transmissions.insert(transmission_id, (transmission, certificate_ids)) {
-                        error!("Failed to insert transmission {transmission_id} into storage - {e}");
-                        continue 'outer;
+
+                        // Also, insert into the cache.
+                        self.cache_transmissions.write().put(transmission_id, (transmission, certificate_ids));
+                    } else if !aborted_transmission_ids.contains(&transmission_id) {
+                        // If the transmission is not found in either storage or the missing map,
+                        // and it's not an aborted transmission, log an error.
+                        error!("Failed to provide a missing transmission {transmission_id}");
                     }
                 }
                 Err(e) => {
+                    // Handle any errors during the retrieval.
                     error!("Failed to process the 'insert' for transmission {transmission_id} into storage - {e}");
-                    continue 'outer;
                 }
             }
         }
-        // Inserts the aborted transmission IDs.
+
+        // Next, handle the aborted transmission IDs.
         for aborted_transmission_id in aborted_transmission_ids {
-            // Retrieve the transmission entry.
             match self.aborted_transmission_ids.get_confirmed(&aborted_transmission_id) {
```

Comment:
this is not coming from this PR, but I think it would be nice to squeeze this block by returning `certificate_ids` from its branches and performing the insertion to persistent storage and its cache below (which would deduplicate those calls)